File: Apps/BankCore/backend/src/main/java/org/d3javu/backend/model/business/client/BusinessClient.java
package org.d3javu.backend.model.business.client;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name="business_client")
@NoArgsConstructor
@Getter
@Setter
public class BusinessClient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "official_name")
    private String officialName;

    @Column(name = "brand")
    private String brand;

    @Column(name = "email")
    private String email;

    @Column(name = "password_hash")
    private String passwordHash;


}

File: Apps/AuthMS/model/business_client/business_client.go
package business_client

type BusinessClient struct {
	id           string
	officialName string
	brand        string
	email        string
	passwordHash string
}

func NewBusinessClient(id string, officialName string, brand string, email string, passwordHash string) *BusinessClient {
	return &BusinessClient{
		id:           id,
		officialName: officialName,
		brand:        brand,
		email:        email,
		passwordHash: passwordHash,
	}
}

func (b *BusinessClient) Id() string           { return b.id }
func (b *BusinessClient) OfficialName() string { return b.officialName }
func (b *BusinessClient) Brand() string        { return b.brand }
func (b *BusinessClient) Email() string        { return b.email }
func (b *BusinessClient) PasswordHash() string { return b.passwordHash }
